packages/components/organisms/f-checkout/src/components/tests/Checkout.test.js

```diff
@@ -619,38 +754,210 @@ describe('Checkout', () => {
             });
         });
 
-        describe('isValidPhoneNumber ::', () => {
-            const isValidPhoneNumberSpy = jest.spyOn(validations, 'isValidPhoneNumber');
+        describe('isFormValid ::', () => {
+            let touchSpy;
+
+            beforeEach(() => {
+                touchSpy = jest.spyOn($v, '$touch');
+            });
+
+            afterEach(() => {
+                jest.clearAllMocks();
+            });
+
+            it('should call `.touch()` ', () => {
+                // Act
+                const wrapper = mount(VueCheckout, {
+                    store: createStore(),
+                    i18n,
+                    localVue,
+                    propsData,
+                    mocks: { $v }
+                });
+
+                wrapper.vm.isFormValid();
+
+                // Assert
+                expect(touchSpy).toBeCalled();
+            });
+
+            it('should return `true` if `$v` is valid', () => {
+                // Arrange
+                $v.$invalid = false;
+
+                // Act
+                const wrapper = mount(VueCheckout, {
+                    store: createStore(),
+                    i18n,
+                    localVue,
+                    propsData,
+                    mocks: { $v }
+                });
+
+                // Assert
+                expect(wrapper.vm.isFormValid()).toBeTruthy();
+            });
+
+            it('should return `false` if `$v` is invalid', () => {
+                // Arrange
+                $v.$invalid = true;
 
-            it('should call `isValidPhoneNumber` from `f-services', () => {
                 // Act
-                mount(VueCheckout, {
+                const wrapper = mount(VueCheckout, {
+                    store: createStore(),
+                    i18n,
+                    localVue,
+                    propsData,
+                    mocks: { $v }
+                });
+
+                // Assert
+                expect(wrapper.vm.isFormValid()).toBeFalsy();
+            });
+        });
+
+        describe('onFormSubmit ::', () => {
+            let isFormValidSpy;
+            let submitCheckoutSpy;
+
+            beforeEach(() => {
+                isFormValidSpy = jest.spyOn(VueCheckout.methods, 'isFormValid');
+                submitCheckoutSpy = jest.spyOn(VueCheckout.methods, 'submitCheckout');
+            });
+
+            afterEach(() => {
+                jest.clearAllMocks();
+            });
+
+            it('should emit `CheckoutFailure` with validation state if form is invalid', async () => {
```

Comment:
Maybe for another PR but i think we could split this test out. There's a few expects in this particular IT statement which indicate it could be broken down a little. Or maybe the first two spy checks aren't really needed?

Reply:
Ahh good point. I've removed the spy checks 👍